BackEnd/profiles/serializers.py

```diff
@@ -378,3 +381,31 @@ def get_user_profile_name(self, obj) -> str:
 
     def get_company_name(self, obj) -> str:
         return obj.company_name
+
+
+class ProfileModerationSerializer(serializers.Serializer):
+    uid = serializers.CharField()
+    action = serializers.CharField()
+    timestamp = serializers.IntegerField()
+
+    def validate_uid(self, value):
+        try:
+            Profile.objects.get(pk=decode_uid(value))
+        except Profile.DoesNotExist:
+            raise serializers.ValidationError("Profile does not exist")
+        return value
+
+    def validate_action(self, value):
+        if value not in ["approve", "reject"]:
+            raise serializers.ValidationError("Action is not allowed")
+        return value
+
+    def validate_timestamp(self, value):
+        try:
+            profile = Profile.objects.get(
+                pk=decode_uid(self.initial_data.get("uid"))
+            )
+            validate_url(profile, value)
+        except ValidationError as error:
```

Comment:
Please use or serializers.ValidationError or ValidationError.

Reply:
try/except block removed, it was redundant